app/gui2/src/stores/graph/index.ts

```diff
@@ -9,7 +9,9 @@ import {
 } from '@/stores/graph/imports'
 import { useProjectStore } from '@/stores/project'
 import { useSuggestionDbStore } from '@/stores/suggestionDatabase'
-import { DEFAULT_VISUALIZATION_IDENTIFIER } from '@/stores/visualization'
+// FIXME: For some reason this is required, otherwise `compilerMessaging.ts` erros with:
+// Cannot access 'VisualizationContainer' before initialization
+import '@/stores/visualization'
```

Comment:
There is a possible dependency cycle:
`@/components/VisualizationContainer.vue` <- `@/providers/visualizationConfig` <- `@/stores/visualization/compilerMessaging` <- `@/components/VisualizationContainer.vue`

Some of the imports in this chain are `import type`, but it seems like those still matter? If it does, then the first loaded module in this chain will determine which modules see which other ones fully defined during synchronous execution. You could try moving the `URLString` type into separate (util?) module, as it is the most common import and that would break the chain.

If this doesn't work, let's try to debug what's going on here. I'm worried that it can behave differently in production build, and cause the electron package to unexpectedly fail to load again.

Reply:
man that was a pain... the issue was `VizContainer` -> `VizSelector` -> `vizStore`. seems like it might be quite painful to fix too...